### PR TITLE
Rejit on lossy float64 to int32 conversion

### LINKED ISSUE #6325: Type confusion between integer and float in JIT
PoC:
```
var b = 0
function test0() {
  var a = 100;

  for (var i = 1; (a |= 1) && i < 6; ++i) {
    test0a();
    for (var j = 0; j < 1; ++j) {
      a = 0xffffffffff
      for (var k = 0; k < 1; ++k) {
        i = a;
        +i;
      }
    }
      print(i) // i should be 1099511627775 (0xffffffffff) but print -1 sometimes
      if(i == 0xffffffffff)
        exit() // Intentionally made ReferenceError to stop
  }

  function test0a() {
    a;
    b = b + 1;
    if(b > 1000)
      return // to avoid out-of-stack
    return test0(b);
  }
}

test0();
```
stdout in Release version:
```
-1  
1099511627775
ReferenceError: 'exit' is not defined
```
`i` should be always 1099511627775 (0xffffffffff) but print -1
It also hits assertion in lib/Backend/GlobOpt.cpp:11940.

Version=v1.11.14
OS=Ubuntu 16.04.6
Arch=x86_64

### PR DESCRIPTION
**Pull request by zenparsing**
Fixes #6325 

In the POC (which repros 1.11 but requires some flags to repro in master), jitting a recursive loop body creates a situation where the type specialization of a sym in the loop prepass does not match the value types that are determined inside of the loop, leading to a lossy conversion from float64 to int32 on the backedge.

This change detects the attempted lossy conversion and rejits with aggressive int type specialization disabled.

### CODE REVIEW

File: test/Optimizer/bug_gh6325.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+var b = 0;
```
Comment (akroshg) [inline test/Optimizer/bug_gh6325.js]: you need to add copyright banner.

File: lib/Backend/GlobOpt.cpp [hunk 2]
```diff
@@ -12163,7 +12163,25 @@ GlobOpt::ToTypeSpecUse(IR::Instr *instr, IR::Opnd *opnd, BasicBlock *block, Valu
...
+                    // and try again.
+                    if (!DoAggressiveIntTypeSpec())
+                    {
+                        if (PHASE_TRACE(Js::BailOutPhase, this->func))
+                        {
+                            Output::Print(_u("    Aborting JIT: AggressiveIntTypeSpec is already disabled\n"));
+                            Output::Flush();
+                        }
+                        throw Js::OperationAbortedException();
+                    }
```
Comment (LouisLaf) [inline lib/Backend/GlobOpt.cpp]: Might as well just use an Assert or an AssertOrFailFast here.

Comment (zenparsing) [reply lib/Backend/GlobOpt.cpp]: Fixed.

**Comment by rhuanjl** [comment]
@zenparsing ok for me to go ahead and merge this based on reviews above?

**Comment by zenparsing** [comment]
@rhuanjl yes

**Pull request merged**